### ISSUE TITLE
Changing learning rate 

### ISSUE THREAD

**Issue by kadir-gunel** [amunmt/amunmt#65]
Hi,

I tried to continue training with a different learning rate. 

I set the -e [finish after this many epoch] parameter to 14 but it continued to train until I manually quit (epoch 16).  And when I reload the model to continue with a different learning rate it started from epoch 8.  Does this happen because it saves the best model ?

And after I quit, the train.log file vanished. 

Also, now, even though I have set the --layer-normalization flag, on the training screen I see it as set to false.

**Comment by kadir-gunel** [comment]
This is serious. I cannot determine the epoch number that I want and sometimes the train.log file is not created. 
Any help ?

**Comment by emjotde** [comment]
I will take a look at it, sounds buggy, would not be surprised if I messed up the epoch counter, but give a little time. 

The log file issue is weird though. I have never seen a log file to dissappear, and I am using the constantly.

**Comment by kadir-gunel** [comment]
Well, may be it did not even created at first time and I did not notice that because I was assuming .log is created during the creation of vocabulary files.

**Comment by emjotde** [comment]
Did you specify --log train.log ? The log should appear right away, but it needs to have the flag, this is not on by default.

**Comment by kadir-gunel** [comment]
Yes, actually I am using the examples run.sh file.

**Comment by emjotde** [comment]
You did not run clean-me.sh by accident?

**Comment by kadir-gunel** [comment]
no. I checked it it was about cleaning bpe folder.

**Comment by kadir-gunel** [comment]
I restarted and also shutdown the computer but no. It still behaves the same.

**Comment by emjotde** [comment]
no log file? How about access rights to that folder?

**Comment by kadir-gunel** [comment]
Well, it is under my home folder.

**Comment by kadir-gunel** [comment]
@emjotde I solved it ; partially. The reason why I do not get .log file was because of a 1 space char between '\' char and the parameters value.
 
But on the other hand, I still cannot determine the epoch number.

**Comment by kadir-gunel** [comment]
After carefully checking all the space chars and eliminate if a space char occur more than one time, everything seems to be working smoothly.

**Comment by emjotde** [comment]
Great. Could you post here an example of what caused the problem?

**Comment by kadir-gunel** [comment]
The below snippet (taken from /examples/training/run.sh since I am modifying that file) is the configuration for training  RO-EN translation system:  

    ../../build/marian \
        --model model/model.npz \
        --devices $GPUS --seed 0 \
        --train-sets data/corpus.bpe.ro data/corpus.bpe.en  \
        --vocabs model/vocab.ro.yml model/vocab.en.yml \
        --dim-vocabs 66000 50000 \
        --dynamic-batching -w 3000 \
        --after-epochs 30 \
        --layer-normalization --dropout-rnn 0.2 --dropout-src 0.1 --dropout-trg 0.2 \
        --early-stopping 5 --moving-average \
        --valid-freq 10000 --save-freq 10000 --disp-freq 1000 \
        --valid-sets data/newsdev2016.bpe.ro data/newsdev2016.bpe.en \
        --valid-metrics cross-entropy valid-script \
        --valid-script-path ./scripts/validate.sh \
        --log model/train.log --valid-log model/valid.log

The problem I was facing was when I try to run the script it wasn't running as intended; like running with default parameters or  not producing .log files.

After long search I found that the problem was caused by spaces! 

If you look really careful to the configurations, you will see that at the end of the --train-set parameter has 2 spaces before '\'. This was causing really big trouble.

After I set each space between parameter value and the '\' char to one space the issue solved.

I hope it was clear.

**Comment by kadir-gunel** [comment]
Oh, there shouldn't be any spaces after `\` too.

**Comment by emjotde** [comment]
Are those spaces in the original script?

**Comment by kadir-gunel** [comment]
No. I was writing quickly and did not care too much about 1-2 spaces.

**Comment by emjotde** [comment]
OK. Thanks. The spaces after \ mess with bash, so that would be expected to cause errors. It is interesting though that those two spaces after --train-sets would cause problems. Maybe boost-program-options thinks there is an empty argument as it expects a lists.

**Comment by kadir-gunel** [comment]
Could you reproduce it ?
